shapely/tests/geometry/test_collection.py

```diff
@@ -38,7 +29,24 @@ def test_empty(geom):
     assert geom.type == geom.geom_type
     assert geom.is_empty
     assert len(geom.geoms) == 0
-    assert geom.geoms == []
+    assert list(geom.geoms) == []
+
+
+def test_empty_subgeoms():
+    geom = shape(
```

Comment:
Seems like this could be somewhat more succinct as:

```Python
geom = GeometryCollection([Point(), LineString()])
```

(not seeing the added value of constructing from GeoJSON here since the focus is on the `.geoms` attribute)